File: week_02/day_2/multiple_classes_lab/bus_lab_start_code/src/bus.py
class Bus:
    def __init__(self, route_number, destination):
        self.route_number = route_number
        self.destination = destination
        self.passengers = []
    
    def drive(self):
        return "Brum brum"
    
    def passenger_count(self):
        return len(self.passengers)
    
    def pick_up(self, person):
        self.passengers.append(person)

    def drop_off(self, person):
        self.passengers.remove(person)

    def empty_bus(self):
        self.passengers.clear()

    def pick_up_from_stop(self, bus_stop):
        bus_stop_queue = bus_stop.queue
        for passenger in bus_stop_queue:
            self.passengers.append(passenger)
        bus_stop.clear_queue()    
        return len(bus_stop_queue)
        

File: week_12/day_2/LabBusStop/src/main/java/Bus.java
import java.util.ArrayList;

public class Bus {
    private String destination;
    private Integer capacity;
    private ArrayList<Person> passengers;

    public Bus(String destination, Integer capacity){
        this.destination = destination;
        this.capacity = capacity;
        this.passengers = new ArrayList<Person>();
    }

    public int busCount() {
        return this.passengers.size();
    }

    public void addPassenger(Person person) {
        if(passengers.size() < capacity){
            this.passengers.add(person);
        }
    }

    public void pickUpPassengerFromStop(BusStop busStop){
        if(passengers.size() < capacity){
            Person person = busStop.removePassenger();
            passengers.add(person);
        }
    }

    public Person removePassenger(){
        return this.passengers.remove(0);
    }
}
